MSGraphSDK/MSGraphSDKTests/MSCollectionRequestTests.m

```diff
@@ -0,0 +1,175 @@
+//
+//  MSCollectionRequestTests.m
+//  MSGraphSDK
+//
+//  Created by canviz on 6/1/16.
+//  Copyright © 2016 Microsoft. All rights reserved.
+//
+
+#import <XCTest/XCTest.h>
+#import "MSGraphTestCase.h"
+@interface MSCollectionRequestTests : MSGraphTestCase
+
+@end
+
+@implementation MSCollectionRequestTests
+
+- (void)setUp {
+    [super setUp];
+    // Put setup code here. This method is called before the invocation of each test method in the class.
+}
+
+- (void)tearDown {
+    // Put teardown code here. This method is called after the invocation of each test method in the class.
+    [super tearDown];
+}
+- (void)testMSCollectionRequest{
+    MSCollectionRequest *collectionRequst = [[MSCollectionRequest alloc] initWithURL:self.testBaseURL client:self.mockClient];
+    XCTAssertNotNil(collectionRequst);
+    
+    
+    NSDictionary *firstObject = @{@"foo" : @"Bar" };
+    NSDictionary *secondObject = @{@"baz" : @"qux" };
+    NSArray *values = @[firstObject, secondObject];
+    NSDictionary *responseObject = @{ MSCollectionValueKey : values, MSODataNextContext : @"foo" };
+    
+    NSHTTPURLResponse *OKresponse = [[NSHTTPURLResponse alloc] initWithURL:self.testBaseURL statusCode:MSExpectedResponseCodesOK HTTPVersion:@"foo" headerFields:nil];
+    [self dataTaskCompletionWithRequest:self.requestForMock data:[NSJSONSerialization dataWithJSONObject:responseObject options:0 error:nil] response:OKresponse error:nil];
+    
+    MSURLSessionDataTask *dataTask = [collectionRequst collectionTaskWithRequest:self.requestForMock odObjectWithDictionary:^MSObject *(NSDictionary *response) {
+        XCTAssertNotNil(response);
+        XCTAssert([values containsObject:response]);
+        MSObject *cast = [[MSObject alloc] initWithDictionary:response];
+        return cast;
+        
+    } completion:^(MSCollection *response, NSError *error) {
+        XCTAssertNotNil(response);
+        XCTAssertNil(error);
+        
+        XCTAssertNotNil(response.value);
+        [response.value enumerateObjectsUsingBlock:^(id obj, NSUInteger idx, BOOL * stop) {
+            XCTAssert([obj isKindOfClass:[MSObject class]]);
+            NSDictionary *result = [obj dictionaryFromItem];
+            XCTAssert([values containsObject:result]);
```

Comment:
This seems to make assumptions about how dictionaryFromItem is implemented, should perform value-wise comparison rather than object-wise.